### ISSUE TITLE
[no-unused-vars] False positive when type is used in declaration

### ISSUE THREAD

**Issue by dbartholomae** [typescript-eslint/typescript-eslint#2510]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  extends: [
    "plugin:@typescript-eslint/recommended"
  ],
  parser: "@typescript-eslint/parser"
}
```

```TS
import { MarkdownChildren } from "./model";

declare global {
  namespace JSX {
    interface IntrinsicElements {
      mdFragment: {
        children?: MarkdownChildren;
      };
    }
  }
}
```

**Expected Result**

This should lint just fine.

**Actual Result**

```
warning  'MarkdownChildren' is defined but never used  @typescript-eslint/no-unused-vars
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3,10,1` |
| `@typescript-eslint/parser`        | `3.10.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `12.16.0` |

**Comment by bradzacher** [comment]
Please. The issue template exists to help you out here.

> I have updated to the latest version of the packages.

You ticked this box, but you listed `3.10.1` as your version. ***This is not the latest version***

**Comment by dbartholomae** [comment]
Oh, sorry, I only tried latest stable version. This is indeed working with `3.10.2-alpha.16` :)
Thanks! I will be more careful next time!

**Comment by dbartholomae** [comment]
Even worse - somehow my tooling fooled me into not seeing version `4.1.0`. Sorry again. I do get another error now, though. Will also check with the current alpha and open another issue if it persists.
